pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/PersistentTopics.java

```diff
@@ -105,11 +105,23 @@ public void getList(@Suspended final AsyncResponse asyncResponse, @PathParam("pr
                     + "namespace level combined (union) with any eventual specific permission set on the topic.")
     @ApiResponses(value = {@ApiResponse(code = 403, message = "Don't have admin permission"),
             @ApiResponse(code = 404, message = "Namespace doesn't exist")})
-    public Map<String, Set<AuthAction>> getPermissionsOnTopic(@PathParam("property") String property,
-            @PathParam("cluster") String cluster, @PathParam("namespace") String namespace,
-            @PathParam("topic") @Encoded String encodedTopic) {
-        validateTopicName(property, cluster, namespace, encodedTopic);
-        return internalGetPermissionsOnTopic();
+    public void getPermissionsOnTopic(@Suspended AsyncResponse asyncResponse,
+                                                              @PathParam("property") String property,
+                                                              @PathParam("cluster") String cluster,
+                                                              @PathParam("namespace") String namespace,
+                                                              @PathParam("topic") @Encoded String encodedTopic) {
+        try {
+            validateTopicName(property, cluster, namespace, encodedTopic);
+            internalGetPermissionsOnTopic().thenAccept(permissions -> asyncResponse.resume(permissions))
+                    .exceptionally(ex -> {
+                        Throwable realCause = FutureUtil.unwrapCompletionException(ex);
+                        log.error("[{}] Failed to get permissions for topic {}", clientAppId(), topicName, ex);
+                        return handleCommonRestAsyncException(asyncResponse, realCause);
```

Comment:
You can use ``resumeAsyncResponseExceptionally`` to instead of ``handleCommonRestAsyncException``. because ``handleCommonRestAsyncException`` will be removed.